src/domain/users/users.repository.ts

```diff
@@ -46,4 +52,116 @@ export class UsersRepository implements IUsersRepository {
       },
     );
   }
+
+  async getUser(authPayload: AuthPayload): Promise<{
+    id: User['id'];
+    status: User['status'];
+    wallets: Array<{ address: Wallet['address']; id: Wallet['id'] }>;
+  }> {
+    if (!authPayload.signer_address) {
+      throw new UnauthorizedException();
+    }
+
+    return await this.postgresDatabaseService.transaction(
+      async (entityManager: EntityManager) => {
+        const walletRepository = entityManager.getRepository(Wallet);
+
+        const authenticatedWallet = await walletRepository.findOne({
+          where: { address: authPayload.signer_address },
+          relations: { user: true },
+        });
+
+        if (!authenticatedWallet?.user) {
+          throw new NotFoundException('User not found');
+        }
+
+        const wallets = await walletRepository.findBy({
+          user: authenticatedWallet.user,
+        });
+
+        return {
+          id: authenticatedWallet.user.id,
+          status: authenticatedWallet.user.status,
+          wallets: wallets.map((wallet) => ({
+            id: wallet.id,
+            address: wallet.address,
+          })),
+        };
+      },
+    );
+  }
+
+  async deleteUser(authPayload: AuthPayload): Promise<void> {
+    if (!authPayload.signer_address) {
+      throw new UnauthorizedException();
+    }
+
+    await this.postgresDatabaseService.transaction(
+      async (entityManager: EntityManager) => {
+        const walletRepository = entityManager.getRepository(Wallet);
+        const userRepository = entityManager.getRepository(DbUser);
+
+        const authenticatedWallet = await walletRepository.findOne({
+          where: { address: authPayload.signer_address },
+          relations: { user: true },
+        });
+
+        if (!authenticatedWallet?.user) {
+          throw new NotFoundException('User not found');
+        }
+
+        // TODO: Check that it also deletes all wallets of that user
+        await userRepository.delete(authenticatedWallet.user);
+      },
+    );
+  }
+
+  async deleteWalletFromUser(args: {
```

Comment:
What do you think about splitting this function a bit as there's quite a bit going on? e.g. `removeWalletFromUser` and `deleteWallet`